### PR TITLE
Demo srs

### PR DESCRIPTION
**Pull request by AxelGard**
Was for DEMO for SRS but became integration and mix of testing

### CODE REVIEW

**Comment by AxelGard** [comment]
@wikstorm Tests seem to run forever. hmmm :thinking:

**Comment by AxelGard** [comment]
and we need to fix Pylint....

**Comment by wikstorm** [comment]
> and we need to fix Pylint....

Pylint was fixed for parts which we were creators of, might assign @aannalarsson and @Nibor9198  for CameraFilter and @PeterWickenberg and @zolo888 for AutopilotAdapter. I'm looking into test stuck.

**Comment by AxelGard** [comment]
Okej pylint and test should be fixed. Some not-needed fixes are left in pylint@wikstorm  @emilcw .

ready for review @aannalarsson @PeterWickenberg :smiley:

File: vgc/pipeline.py [hunk 1]
```diff
@@ -1,50 +1,84 @@
...
+# pylint: disable=too-many-arguments
+# 6 arguments are needed.
+
 class Pipeline:
     """"
     Cross thread messaging pipeline.
     Threads communicates by calling pipeline's
     class functions
 
-    __init__(self):
```
Comment (PeterWickenberg) [inline vgc/pipeline.py:14]: Should there really be a description of how the class functions are used here? The same description is writen under all functions

File: vgc/view_controller_fast.py [hunk 2]
```diff
@@ -157,11 +165,12 @@ def update_server_input(self, theta = 0, phi = 0, lock_on = False, zoom_in = 2):
...
+            if(lock_on is True and self.lock_on_in is False):
                 self.init_lock_on = True
-            self.lock_on = lock_on
+            self.lock_on_in = lock_on
             if(zoom_in >= 2 and zoom_in <= 50):
                 self.camera_zoom = zoom_in
+            self.server_write = False
 
 
     def deg2rad(self, degrees):
```
Comment (PeterWickenberg) [inline vgc/view_controller_fast.py:176]: 0.017453295 should be a defined constant

File: vgc/view_controller_fast.py [hunk 3]
```diff
@@ -186,11 +195,16 @@ def coordinate_to_point(self, coord1, coord2, height):
...
+        if(x_diff == 0 and y_diff == 0):
+            phi = 0
+        else:
+            phi = np.arctan2(x_diff, y_diff)
         theta_2 = tf.arctan((((x_diff**2) + (y_diff**2))**0.5) / height)
-        return self.rad2deg(theta_2), (self.rad2deg(phi) + 360) % 360
+        return self.rad2deg(theta_2), self.rad2deg(phi) % 360
 
     def point_to_coordinate(self, theta, phi, height, d_coord):
         """
```
Comment (PeterWickenberg) [inline vgc/view_controller_fast.py:210]: "our" aim? and "our" height? should be systems height or something, also, can't quite understand last sentence

File: vgc/view_controller_fast.py [hunk 4]
```diff
@@ -235,38 +249,51 @@ def main(self, is_threading=False, debug=True):
...
+                self.d_roll = self.d_roll_in
+                self.d_pitch = self.d_pitch_in
+                self.d_yaw = self.d_yaw_in
+                self.d_height = self.d_height_in
+                self.d_coordinate = self.d_coordinate_in
+                self.autopilot_write = False
+
+            if self.lock_on:
+                if self.init_lock_on:
+                    #Phi = 0 ska ge samma latitud, och en annan longitud
```
Comment (PeterWickenberg) [inline vgc/view_controller_fast.py:269]: No svenski please

**Review by PeterWickenberg — CHANGES_REQUESTED** [review]
I didn't find that many problems, only a cupple comments here and there. It aint much, but it's honest work.

**Comment by AxelGard** [comment]
@PeterWickenberg  all good?

**Comment by PeterWickenberg** [comment]
Its ok

**Comment by AxelGard** [comment]
Don't close,  approve changes so it can be merged

**Review by PeterWickenberg — APPROVED** [review]
Like so?

**Pull request merged**